src/main/java/webserver/data/TodosHelper.java

```diff
@@ -0,0 +1,28 @@
+package webserver.data;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import webserver.models.Todo;
+
+public class TodosHelper {
+    public static List<Todo> getFilteredTodos(String query, List<Todo> todos) { 
+        List<Todo> filteredTodos = new ArrayList<>();
+        
+        for (Todo todo : todos) {
+            String title = todo.getTitle().toLowerCase();
+            String text = todo.getText().toLowerCase();
+            if (title.contains(query) || text.contains(query)) {
+                filteredTodos.add(todo);
+            }
+        }
+        return filteredTodos;
+    }
+
+    public static Todo getTodo(int id, List<Todo> todos) {
```

Comment:
This might be a good place to return `Optional<Todo>` so you don't have to default to `null`.

Reply:
Oh yes! I used it in my router. Will look into it more.